packages/spacecat-shared-data-access/src/models/api-key/api-key.js

```diff
@@ -40,8 +40,37 @@ const ApiKey = (data) => {
   self.getCreatedAt = () => self.state.createdAt;
   self.getExpiresAt = () => self.state.expiresAt;
   self.getRevokedAt = () => self.state.revokedAt;
+  self.getDeletedAt = () => self.state.deletedAt;
   self.getScopes = () => self.state.scopes;
 
+  /**
```

Comment:
Would this model be the right place for an `isValid()` method, which could check all the internal properties to ensure that a key is still valid?

Reply:
Or `isActive()`, alternatively?

Reply:
We could do that in the service. Do we need it in spacecat-shared?

Reply:
If we do so in the service than anywhere that re-uses this model will need to implement something similar. I see it as a direct concern of the model itself, "am I still valid", and think that it makes sense to have that logic live alongside the model. 

I also like the pattern of "asking" the entity itself, are you still valid: `if (apiKeyInstance.isValid()) {` — nice and clean.